### ISSUE TITLE
Just work at the first time. 

### ISSUE THREAD

**Issue by ltkun** [matiasdelellis/facerecognition#485]
I have passed the test by running occ face:background_job. But after uploading several photos the command doesn't work anymore. No errors. Here is the output:
 

root@nextcloud ~# sudo -u www-data php -f /var/www/nextcloud/occ face:background_job
PHP Warning:  Module 'pdlib' already loaded in Unknown on line 0
1/10 - Executing task CheckRequirementsTask (Check all requirements)
2/10 - Executing task CheckCronTask (Check that service is started from either cron or from command)
3/10 - Executing task LockTask (Acquire lock so that only one background task can run)
4/10 - Executing task DisabledUserRemovalTask (Purge all the information of a user when disable the analysis.)
5/10 - Executing task StaleImagesRemovalTask (Crawl for stale images (either missing in filesystem or under .nomedia) and remove them from DB)
6/10 - Executing task CreateClustersTask (Create new persons or update existing persons)
        Skipping cluster creation, not enough data (yet) collected. For cluster creation, you need either one of the following:
        * have 1000 faces already processed
        * or you need to have 95% of you images processed
        Use stats command to track progress
        Clusters already exist, estimated there is no need to recreate them
        Skipping cluster creation, not enough data (yet) collected. For cluster creation, you need either one of the following:
        * have 1000 faces already processed
        * or you need to have 95% of you images processed
        Use stats command to track progress
        Skipping cluster creation, not enough data (yet) collected. For cluster creation, you need either one of the following:
        * have 1000 faces already processed
        * or you need to have 95% of you images processed
        Use stats command to track progress
7/10 - Executing task AddMissingImagesTask (Crawl for missing images for each user and insert them in DB)
        Skipping image scan for user admin that has disabled the analysis
        Skipping image scan for user ly that has disabled the analysis
        Skipping image scan for user wjw that has disabled the analysis
8/10 - Executing task EnumerateImagesMissingFacesTask (Find all images which don't have faces generated for them)
9/10 - Executing task ImageProcessingTask (Process all images to extract faces)
        NOTE: Starting face recognition. If you experience random crashes after this point, please look FAQ at https://github.com/matiasdelellis/facerecognition/wiki/FAQ
10/10 - Executing task UnlockTask (Release obtained lock)

And if I reset the results,do background_job again the new photo will be found.

**Comment by matiasdelellis** [comment]
Hi @ltkun 
What version of nextcloud do you have? what type of storage do you use? and how do you upload the images?

**Comment by ltkun** [comment]
> Hi @ltkun 
> What version of nextcloud do you have? what type of storage do you use? and how do you upload the images?

I use a lxc nextcloud 21.0.2 in PVE. And other function works.   The storage is in default directory /var/www/nextcloud/data/ in a vm disk. I have uploaded the images from browser. I have had this problem since I installed facerecognition. Thanks!

**Comment by harakiwi1** [comment]
Hello,
I've encountered the same issue and I was able to find kind of workaround using `./occ face:reset --image-errors`. I'm running dockerized Nextcloud 21.0.2.
1. I've added about 2000 images directly to /admin/data/Photos folder
2. I've run `occ files:scan --all` and images appeared in Nextcloud web
3. I've run `occ face:background_job` and all images were successfully processed
4. I've repeated steps 1-3 with more photos, but in step 3 no new images were processed
5. I've uploaded 1 new image via web interface
6. I've run `occ face:background_job` and only the 1 image was processed
7. I've run `./occ face:reset --image-errors`
8. I've run `occ face:background_job` and suddenly all the photos were processed

So @ltkun as an workaround I suggest to run ./occ face:reset --image-errors` and then start the background job again.

**Comment by ltkun** [comment]
@harakiwi1 thanks I will try it later.

**Comment by matiasdelellis** [comment]
Hi @ltkun @harakiwi1 

I have some doubts regarding this yet.. 🤔 

On the one hand, clarify how it should work.

1. You add about 2000 images from some client.
2. Run occ face:background_job and all images were successfully processed.
3. Run occ face:background_job again to clustering the faces. (Or you can use the ´--defer-clustering´ on step 2)
4. Upload 1 new image via web interface.
5. Run occ face:background_job again...

*NOTE:* At this point, the process should analyze all 5 photos. If it fails, it's wrong.. 😞 
If you run the command with debug you should see something like this:
´´´
[matias@nube nextcloud]$ time sudo -u apache php occ face:background_job -u matias -vvv
1/10 - Executing task CheckRequirementsTask (Check all requirements)
	System: Linux
	System memory: 7804407808
	PHP Memory Limit: Unknown
..................
6/10 - Executing task CreateClustersTask (Create new persons or update existing persons)
	Found 6 faces without associated persons for user matias and model 4
	Oldest face without persons for user matias and model 4 is from 2021-06-10 12:20:20
	Clusters already exist, estimated there is no need to recreate them
´´´

In summary, when you add new images, it must detect it, and analyze them, but the face clustering is made maximum every 2 hours. This is done on purpose to avoid continual recluster them that is a very expensive process.

So if you keep running the command, you will see something like the above.. When at least two hours have passed since the last clustering, they will finally clustering again.

Again, if the images are not detected, and they are not analyzed, there is something wrong, but if they are simply not cluster you must wait two hours to clustering the new faces. **Please test and confirm**

Remember that the application is intended for continuous execution. Then you should not worry about the result of a simple execution. 😅
